src/encoder.rs

```diff
@@ -3492,22 +3492,33 @@ pub fn encode_frame<T: Pixel>(
   }
   let tile_group = encode_tile_group(fi, fs, inter_cfg);
 
-  write_obus(&mut packet, fi, fs, inter_cfg).unwrap();
+  if fi.frame_type == FrameType::KEY {
+    write_key_frame_obus(&mut packet, fi, obu_extension).unwrap();
+  }
   let mut buf1 = Vec::new();
+  let mut buf2 = Vec::new();
+  {
+    let mut bw2 = BitWriter::endian(&mut buf2, BigEndian);
+    bw2.write_frame_header_obu(fi, fs, inter_cfg).unwrap();
+  }
+
   {
     let mut bw1 = BitWriter::endian(&mut buf1, BigEndian);
-    bw1.write_obu_header(ObuType::OBU_TILE_GROUP, 0).unwrap();
+    bw1.write_obu_header(ObuType::OBU_FRAME, obu_extension).unwrap();
   }
   packet.write_all(&buf1).unwrap();
   buf1.clear();
 
   {
     let mut bw1 = BitWriter::endian(&mut buf1, BigEndian);
-    bw1.write_uleb128(tile_group.len() as u64).unwrap();
+    bw1.write_uleb128((buf2.len() + tile_group.len()) as u64).unwrap();
```

Comment:
I can't tell what's going on here. It is writing data from `buf1` but for the length of `buf2`. Why?